control/control_test.go

```diff
@@ -1075,13 +1078,13 @@ func TestCollectDynamicMetrics(t *testing.T) {
 			// pool should be the global cache expiration
 			So(ttl, ShouldEqual, strategy.GlobalCacheExpiration)
 			mts, errs := c.CollectMetrics([]core.Metric{m}, time.Now().Add(time.Second*1), taskID)
-			hits, err := pool.CacheHits(core.JoinNamespace(m.namespace), 2, taskID)
+			hits, err := pool.CacheHits(m.namespace.String(), 2, taskID)
```

Comment:
Should this not just be m.NamespaceAsString() since we have the function? IDK if it really matters...

```go
func (m *metricType) Namespace() core.Namespace {
	return m.namespace
}

func (m *metricType) NamespaceAsString() string {
	return m.Namespace().String()
}
```